pkg/asset/machines/master.go

```diff
@@ -181,7 +182,17 @@ func (m *Master) Generate(dependencies asset.Parents) error {
 		mpool.Set(ic.Platform.OpenStack.DefaultMachinePlatform)
 		mpool.Set(pool.Platform.OpenStack)
 		pool.Platform.OpenStack = &mpool
-		machines, err = openstack.Machines(clusterID.InfraID, ic, pool, string(*rhcosImage), "master", "master-user-data")
+
+		imageName := clusterID.InfraID + "-rhcos"
+		// Here we check whether rhcosImage is a url or not. If this is the case, it means that the image was
+		// created by the installer and we have to use the universal name "<infraID>-rhcos". Otherwise, it means
```

Comment:
Nit: `rhcosImage` being an URL doesn't necessary mean it was created by the installer (it's possible that the user gives an URL to `OPENSHIFT_INSTALL_OS_IMAGE_OVERRIDE`) *but* it means it needs to be imported to glance under a unique name, `clusterID.InfraID + "-rhcos"`

Reply:
Not exactly right... If `rhcosImage` is a URL, then the installer always creates an image from this URL with name `clusterID.InfraID + "-rhcos"`. It doesn't matter how the URL was obtained, from `rhcos.json` or from `OPENSHIFT_INSTALL_OS_IMAGE_OVERRIDE` variable.